### PR TITLE
dimred api

### PR DESCRIPTION
**Pull request by bomquangia**

### CODE REVIEW

File: BioTuringSDK/Server/app.py [hunk 1]
```diff
@@ -137,7 +143,25 @@ def create_gene_list(username: str, study_id: str, options: CreateGeneListReques
         col_id = app.server.create_gene_list(username, study_id, options.name, options.genes)
         return col_id
     except Exception as e:
-        raise HTTPException(400, "Unable to import metadata due to error: %s"
+        raise HTTPException(400, "Unable to create gene list due to error: %s"
+                                % helpers.exc_to_str(e))
+
+@app.post("/studies/{username}/{study_id}/add_dimred")
```
Comment (JATC1024) [inline BioTuringSDK/Server/app.py:149]: Please add "/sdk" to the front of the path (for now).

**Review by JATC1024 — CHANGES_REQUESTED** [review]
Please take a look at the comments

File: BioTuringSDK/Server/app.py (hunk 1 above)
Comment (bomquangia) [reply BioTuringSDK/Server/app.py]: Any idea how to write [test_dimred.py](https://github.com/BioTuring-Notebooks/BBrowserSDK/pull/78/files#diff-68950b1272164e3251aa95ce6d50fab636f8536f528e5d4561c57213a1fcf61e) to pass in `user` argument?

File: BioTuringSDK/Server/app.py [hunk 2]
```diff
@@ -53,6 +54,11 @@ class CreateGeneListRequest(BaseModel):
     name: str
     genes: List[str]
 
+class AddDimredRequest(BaseModel):
+    name: str
+    id: Optional[str]
+    coords: List[List[int]]
```
Comment (JATC1024) [inline BioTuringSDK/Server/app.py:60]: Should the coordinates be `float` instead of `int`?

**Review by JATC1024 — DISMISSED** [review]

**Pull request closed without merging**